ops/ccs-ops-misc/securityhub/resolve-inactive-findings/resolve-inactive-findings.py

```diff
@@ -0,0 +1,530 @@
+ACTIVE_VOLUME_STATES = set(EC2_VOLUME_STATES) - set(['deleting', 'deleted', 'error'])
+
+# When we resolve a finding, we add a note to explain why.
+RESOLVED_NOTE = "Finding no longer references active resources."
+RESOLVED_BY = "Script"
+
+# This is the filter we use to find findings that need to be resolved. 
+# The 'ResourceType' gets set via command line args.
+FINDING_FILTERS = {
+    'ResourceType': [],
+    'WorkflowStatus': [
+        {
+            'Value': 'NEW',
+            'Comparison': 'EQUALS'
+        },
+        {
+            'Value': 'NOTIFIED',
+            'Comparison': 'EQUALS'
+        },
+    ],
+    'RecordState': [
+        {
+            'Value': 'ACTIVE',
+            'Comparison': 'EQUALS'
+        },
+    ]
+}
+
+# Resource ID regex patterns
+RESOURCE_ID_RE = {
+    'AwsEc2Instance': r'^i-[a-zA-Z0-9]+$',
+    'AwsS3Bucket': r'^[a-z0-9][a-zA-Z0-9-]{1,61}[a-z0-9]$',
+    'AwsEc2Volume': r'^vol-[a-zA-Z0-9]+$',
+    'AwsEc2SecurityGroup': r'^sg-[a-zA-Z0-9]+$',
+    'AwsRdsDbClusterSnapshot': r'^[a-zA-Z0-9-]+$',
+    'AwsRdsDbInstance': r'^[a-zA-Z0-9-]+$',
+    'AwsIamUser': r'^[a-zA-Z0-9-]+$',
+    'AwsIamPolicy': r'^[a-zA-Z0-9-]+$',
+    'AwsIamAccessKey': r'^AWS::IAM::AccessKey:[A-Z0-9]+$',
+    'AwsAutoScalingAutoScalingGroup': r'^[a-zA-Z0-9-]+$',
+    'AwsLambdaFunction': r'^[a-zA-Z0-9-]+$',
+    'AwsSnsTopic': r'^[a-zA-Z0-9-]+$',
+    'AwsKmsKey': r'^[a-zA-Z0-9-]+$',
+}
+
+# SecurityHub Insights are used to get a count of findings that match our filters. This is much faster than
+# paginating through all the findings as there is no .count() method. Insights are dynamically created if
+# needed.
+INSIGHT_NAME_PREFIX = 'NewFailedFindingsBy'
+GROUP_BY = 'AwsAccountId'
+
+
+# Return a boto3 client based on the resource type or 'hub' for a security hub client
+def get_client(region, resource_type):
+    if resource_type.startswith('AwsEc2'):
+        return boto3.client('ec2', region_name=region)
+    elif resource_type.startswith('AwsS3Bucket'):
+        return boto3.client('s3', region_name=region)
+    elif resource_type.startswith('AwsRds'):
+        return boto3.client('rds', region_name=region)
+    elif resource_type.startswith('AwsIam'):
+        return boto3.client('iam', region_name=region)
+    elif resource_type.startswith('AwsAutoScaling'):
+        return boto3.client('autoscaling', region_name=region)
+    elif resource_type.startswith('AwsLambda'):
+        return boto3.client('lambda', region_name=region)
+    elif resource_type.startswith('AwsSns'):
+        return boto3.client('sns', region_name=region)
+    elif resource_type.startswith('AwsKms'):
+        return boto3.client('kms', region_name=region)
+    elif resource_type == 'hub':
+        return boto3.client('securityhub', region_name=region)
+
+
+# Returns True if the given "id" string looks like an AWS ARN
+def is_arn(id):
+    return id.startswith('arn:')
```

Comment:
I assume that a linter might take issue with overwriting the `<built-in function id>` here and elsewhere. I see you've used `resource_id` in other signatures and `_id` is a common, pragmatic solution to avoiding conflicts with built-ins, too. I don't think there's any danger here, though ;)